Client/Frontend/Browser/OpenSearch.swift

```diff
@@ -78,11 +78,24 @@ class OpenSearchEngine: NSObject, NSCoding {
         // and have to do flaky pattern matching instead.
         let placeholder = "PLACEHOLDER"
         let template = searchTemplate.replacingOccurrences(of: SearchTermComponent, with: placeholder)
-        let components = URLComponents(string: template)
-        let searchTerm = components?.queryItems?.filter { item in
+        var components = URLComponents(string: template)
+        
+        if let retVal = extractQueryArg(in: components, for: placeholder) {
+            return retVal
+        } else {
+            // Query arg may be exist inside fragment
+            components = URLComponents()
+            components?.query = URL(string: template)?.fragment
+            return extractQueryArg(in: components, for: placeholder)
+        }
+    }
+    
+    fileprivate func extractQueryArg(in urlComponents : URLComponents?,
+                                    for placeholder : String) -> String? {
+        let searchTerm = urlComponents?.queryItems?.filter { item in
             return item.value == placeholder
         }
-        guard let term = searchTerm, !term.isEmpty  else { return nil }
+        guard let term = searchTerm, !term.isEmpty else { return nil }
         return term[0].name
     }
```

Comment:
Acting directly on query items is clearer to me:
```
 fileprivate func extractQueryArg(in queryItems: [URLQueryItem]?, for placeholder: String) -> String? {
        let searchTerm = queryItems?.filter { item in
            return item.value == placeholder
        }
        return searchTerm?.first?.name
    }
```

Reply:
Changed.